pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -699,41 +740,24 @@ func (a *AmbientIndex) handlePod(oldObj, newObj any, isDelete bool, c *Controlle
 	a.mu.Lock()
 	defer a.mu.Unlock()
 	updates := sets.New[model.ConfigKey]()
-	// This is a waypoint update
-	if p.Labels[constants.ManagedGatewayLabel] == constants.ManagedGatewayMeshControllerLabel {
-		scope := model.WaypointScope{Namespace: p.Namespace, ServiceAccount: p.Annotations[constants.WaypointServiceAccount]}
-		ip := p.Status.PodIP
-		if isDelete || !IsPodReady(p) {
-			if a.waypoints[scope].Contains(ip) {
-				sets.DeleteCleanupLast(a.waypoints, scope, ip)
-				updates.Merge(a.updateWaypoint(scope, ip, true, c))
-			}
-		} else {
-			if _, f := a.waypoints[scope]; !f {
-				a.waypoints[scope] = sets.New[string]()
-			}
-			if !a.waypoints[scope].InsertContains(ip) {
-				updates.Merge(a.updateWaypoint(scope, ip, false, c))
-			}
-		}
-	}
 
 	var wl *model.WorkloadInfo
 	if !isDelete {
 		wl = c.extractWorkload(p)
 	}
-	oldWl := a.byPod[p.Status.PodIP]
+	oldWl := a.byPod[c.Network(p.Status.PodIP, p.Labels).String()+"/"+p.Status.PodIP]
```

Comment:
same nit about using internal types to key these maps so we get compilation errors rather than bugs if future devs use the wrong key formatting